### PR TITLE
Feat new send feedback modal

### PR DESCRIPTION
**Pull request by Bluedrop**
## Context

Jira Issue:

ISEEC-3040

## Checklist:

- [ ] Has been verified in an integrated environment
- [x] Has relevant unit and integration tests passing
- [x] Has no linting, test console, or browser console errors (best effort)
- [x] Has JSDoc comment blocks for complex code

## PR Review Guidance

The contents might seem tailored to a specific organization, but the platformName is confirable. As well as the submit an idea link.

### CODE REVIEW

File: src/components/Feedback/Feedback.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        text="Submit an Idea"
+        iconName="idea"
+        className={`${prefix}--bmrg-feedback-container`}
+      >
+        {({ closeModal }) => {
+          return (
+            <>
+              <ModalHeader
+                title="Submit an Idea"
+                closeModal={() => this.handleCloseModal(closeModal)}
```
Comment (timrbula) [inline src/components/Feedback/Feedback.js]: I'm guessing this is a carry over from the other code? Can we simplify this by just passing the closeModal function. 
```suggestion
                closeModal={closeModal}
```

File: src/components/UIShell/UIShell.js [hunk 2]
```diff
@@ -257,6 +259,9 @@ function UIShell({
               text="Flow Documentation"
             />
           ),
+          isFeedbackEnabled && (
+            <Feedback platformName={platformName} sendIdeasUrl="https://ideas.ibm.com" key="Feedback" />
```
Comment (timrbula) [inline src/components/UIShell/UIShell.js]: Per comment in Slack, any discussion to make this a dynamic value that comes from the service?

File: src/components/Feedback/Feedback.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+              <ModalBody>
+                <div className={`${prefix}--bmrg-feedback`}>
+                  <p>Share your thoughts and ideas on what we can do to improve the {this.props.platformName} platform and our Onboarding process.</p>
+                  <p>
+                    To submit your idea, visit our portal at
+                    {" " /* We need to force a space before the link tag */}
+                    <a href={this.props.sendIdeasUrl}
+                      target="_blank"
+                      rel="noopener noreferrer"
+                      aria-describedby="new-window-aria-desc-0">
```
Comment (timrbula) [inline src/components/Feedback/Feedback.js]: 👍

**Review by timrbula — COMMENTED** [review]
Overall looks good from a code POV

File: src/components/UIShell/UIShell.js (hunk 2 above)
Comment (Bluedrop) [reply src/components/UIShell/UIShell.js]: Yes I just missed this and left the default value in.

File: src/components/Feedback/Feedback.js (hunk 1 above)
Comment (Bluedrop) [reply src/components/Feedback/Feedback.js]: I thought there's some sort of code styling that is required/suggested to be followed. The simple one looks better to me too.

Comment (timrbula) [reply src/components/Feedback/Feedback.js]: Cool yeah given that we used class components I would guess its pretty old code at this point. I probably didn't know what I was doing really.

File: src/components/Feedback/Feedback.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+              </ModalFooter>
+            </>
+          );
+        }}
+      </HeaderMenuItem>
+    );
+  }
+}
+
+Feedback.propTypes = { baseServiceUrl: PropTypes.string };
```
Comment (BenjaminRuby) [inline src/components/Feedback/Feedback.js]: could we add `sendIdeasUrl` to propTypes?

File: src/components/UIShell/UIShell.js [hunk 5]
```diff
@@ -157,6 +163,7 @@ UIShell.defaultProps = {
   renderPrivacyStatement: true,
   user: {},
   renderRightPanel: {},
+  sendIdeasUrl: "https://ideas.ibm.com",
```
Comment (BenjaminRuby) [inline src/components/UIShell/UIShell.js]: 👍

**Review by BenjaminRuby — CHANGES_REQUESTED** [review]
@Bluedrop just that small adding to the propTypes and should be good to go for me

File: src/components/Feedback/Feedback.js (hunk 4 above)
Comment (Bluedrop) [reply src/components/Feedback/Feedback.js]: Done. I'm so unused to having the props at the end of the file... :)

**Review by BenjaminRuby — DISMISSED** [review]
LGTM 👍

File: src/components/Feedback/_feedback.scss [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      height: rem(448px);
+    }
+
+    .#{$prefix}--bmrg-feedback {
+      color: $bmrg-brand-white;
+      height: rem(100px);
+      margin-top: rem(16px);
+      width: 100%;
+
+      a {
```
Comment (timrbula) [inline src/components/Feedback/_feedback.scss:13]: image

We should use another theme token here for a11y and making sure it works for both themes. For the boomerang theme the `bmrg-tertiary-color` is the same as the background color.

Comment (timrbula) [reply src/components/Feedback/_feedback.scss:13]: I think the `bmrg-header-modal-confirm-color` token would work for both themes. It is the same color as the background of the "OK" button.

**Review by timrbula — APPROVED** [review]

**Pull request merged**